lift/util.h

```diff
@@ -44,11 +49,31 @@ static inline LIFT_HOST_DEVICE void swap(T& a, T& b)
 }
 
 // get the lane ID for a GPU thread
-static inline __device__ unsigned int lane_id(void)
+static inline LIFT_DEVICE unsigned int lane_id(void)
 {
+#if LIFT_DEVICE_COMPILATION
     unsigned int ret;
     asm("mov.u32 %0, %laneid;" : "=r"(ret));
     return ret;
+#else
+    assert(!"lane_id not supported on CPU");
+    return 0;
+#endif
+}
+
+// perform integer division, rounding up
+// result = (a + b - 1) / b
+template <typename T>
+static inline LIFT_HOST_DEVICE T idivup(const T a, const T b)
+{
+    // compile-time type checking: make sure we don't get floats passed in
+    static_assert(std::is_integral<T>::value && !std::is_same<T, bool>::value,
+                  "idivup requires integral numeric type");
+
+    // runtime check for wraparound
+    assert((b > 0 && (a + b - 1) > a) || (b < 0 && (a + b - 1) < a));
```

Comment:
I don't really know of a use case for calling idivup with a negative denominator, but that's just me. If you do keep it in, it would be good to add a test case for it.